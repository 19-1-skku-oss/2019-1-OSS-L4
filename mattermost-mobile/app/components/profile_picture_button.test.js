// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import ProfilePictureButton from './profile_picture_button.js';

import {Client4} from 'mattermost-redux/client';

describe('profile_picture_button', () => {
    const navigator = {
        setOnNavigatorEvent: jest.fn(),
        setButtons: jest.fn(),
        dismissModal: jest.fn(),
        push: jest.fn(),
    };

    const baseProps = {
        theme: Preferences.THEMES.default,
        navigator,
        currentUser: {
            first_name: 'Dwight',
            last_name: 'Schrute',
            username: 'ieatbeets',
            email: 'dwight@schrutefarms.com',
            nickname: 'Dragon',
            position: 'position',
        },
        blurTextBox: jest.fn(),
        maxFileSize: 20 * 1024 * 1024,
        uploadFiles: jest.fn(),
    };

    test('should match snapshot', async () => {
        const wrapper = shallow(
            <ProfilePictureButton {...baseProps}/>,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should NOT return option to remove when profile picture is default', () => {
        Client4.getProfilePictureUrl = jest.fn(() => 'image.png');
        const wrapper = shallow(
            <ProfilePictureButton {...baseProps}/>,
        );
        const instance = wrapper.instance();

        // test default image (WITHOUT query param)
        instance.getRemoveProfileImageOption();
        expect(wrapper.state('extraOptions')).toEqual([null]);
    });

    test('should return option to remove profile picture if customized', () => {
        Client4.getProfilePictureUrl = jest.fn(() => 'image.png?query');
        const wrapper = shallow(
            <ProfilePictureButton {...baseProps}/>,
        );
        const instance = wrapper.instance();

        // test custom image (WITH query param)
        instance.getRemoveProfileImageOption();
        expect(wrapper.state('extraOptions')).not.toEqual([null]);
    });
});
