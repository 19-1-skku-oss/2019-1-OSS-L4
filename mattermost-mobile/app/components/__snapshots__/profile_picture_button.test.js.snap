// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile_picture_button should match snapshot 1`] = `
<AttachmentButton
  blurTextBox={[MockFunction]}
  browseFileTypes="public.item"
  canBrowseFiles={true}
  canBrowsePhotoLibrary={true}
  canBrowseVideoLibrary={true}
  canTakePhoto={true}
  canTakeVideo={true}
  extraOptions={
    Array [
      null,
    ]
  }
  maxFileCount={5}
  maxFileSize={20971520}
  navigator={
    Object {
      "dismissModal": [MockFunction],
      "push": [MockFunction],
      "setButtons": [MockFunction],
      "setOnNavigatorEvent": [MockFunction],
    }
  }
  theme={
    Object {
      "awayIndicator": "#ffbc42",
      "buttonBg": "#166de0",
      "buttonColor": "#ffffff",
      "centerChannelBg": "#ffffff",
      "centerChannelColor": "#3d3c40",
      "codeTheme": "github",
      "dndIndicator": "#f74343",
      "errorTextColor": "#fd5960",
      "linkColor": "#2389d7",
      "mentionBj": "#ffffff",
      "mentionColor": "#145dbf",
      "mentionHighlightBg": "#ffe577",
      "mentionHighlightLink": "#166de0",
      "newMessageSeparator": "#ff8800",
      "onlineIndicator": "#06d6a0",
      "sidebarBg": "#145dbf",
      "sidebarHeaderBg": "#1153ab",
      "sidebarHeaderTextColor": "#ffffff",
      "sidebarText": "#ffffff",
      "sidebarTextActiveBorder": "#579eff",
      "sidebarTextActiveColor": "#ffffff",
      "sidebarTextHoverBg": "#4578bf",
      "sidebarUnreadText": "#ffffff",
      "type": "Mattermost",
    }
  }
  uploadFiles={[MockFunction]}
  validMimeTypes={Array []}
/>
`;
